runtime/compiler/control/J9Options.cpp

```diff
@@ -1609,6 +1613,18 @@ void J9::Options::preProcessCodeCacheIncreaseTotalSize(J9JavaVM *vm, J9JITConfig
    if (!codecachetotalAlreadyParsed) // avoid processing twice for AOT and JIT and produce duplicate messages
       {
       codecachetotalAlreadyParsed = true;
+
+      UDATA ccTotalSize = jitConfig->codeCacheTotalKB;
+      bool incomplete;
+      uint64_t freePhysicalMemory = getCompilationInfo(jitConfig)->computeAndCacheFreePhysicalMemory(incomplete);
+      if (freePhysicalMemory != OMRPORT_MEMINFO_NOT_AVAILABLE && !incomplete && freePhysicalMemory <= 512 * 1024 * 1024)
```

Comment:
We should not blindly assume that the default value for codeCacheTotalKB is 256MB. Rather, use that value as a maximum value when there is plenty of physical memory. So something like:
```
poposedCodeCacheTotalKB = (computeAndCacheFreePhysicalMemory() * TR::Options::getCodeCacheMaxPercentageOfAvailableMemory() / 100) >> 10
if (proposedCodeCacheTotalKB < jitConfig->codeCacheTotalKB)
  ccTotalSize = proposedCodeCacheTotalKB << 10
```
pay attention when the values are expressed in bytes, vs KB.

Reply:
I think we should enter this code block assuming we are running on low memory and hence assuming 512MB (or another value) as a low memory criteria to enter this code block would be nice. I will make the change to introduce the check for `proposedCodeCacheTotalKB < jitConfig->codeCacheTotalKB`. We should have the `ccTotalSize` in KB as we set this value directly to `jitConfig->codeCacheTotalKB`

Reply:
Corrected code to take the defaults for `jitConfig->codeCacheTotalKB` as the value to check the available free memory against.